MSBuildTasks/BuildTestOutputFiles.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.  rubicon licenses this file to you under 
+// the Apache License, Version 2.0 (the "License"); you may not use this 
+// file except in compliance with the License.  You may obtain a copy of the 
+// License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT 
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the 
+// License for the specific language governing permissions and limitations
+// under the License.
+// 
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+namespace Remotion.BuildTools.MSBuildTasks
+{
+  public class BuildTestOutputFiles : Task
+  {
+    [Required] public ITaskItem[] Input { get; set; }
+
+    [Output] public ITaskItem[] Output { get; set; }
+
+    public override bool Execute ()
+    {
+      var output = new List<ITaskItem>();
+      foreach (var item in Input)
+      {
+        var testingConfiguration = item.GetMetadata ("TestingConfiguration");
+
+        var configurations = testingConfiguration.Split (';');
+        var configurationIDIndex = 0;
+        foreach (var configuration in configurations)
+        {
+          var splitConfigurationItems = configuration.Split ('+');
+          var newItem = CreateTaskItem (item, configurationIDIndex, splitConfigurationItems);
+          output.Add (newItem);
+          configurationIDIndex++;
+        }
+      }
+
+      Output = output.ToArray();
+      return true;
+    }
+
+    private ITaskItem CreateTaskItem (ITaskItem originalItem, int configurationIDIndex, IReadOnlyList<string> configurationItems)
+    {
+      var item = new TaskItem (originalItem.ItemSpec);
+      item.SetMetadata (TestingConfigurationMetadata.ID, $"{originalItem.ItemSpec}_{configurationIDIndex}");
+
+      var browser = configurationItems[0];
+      item.SetMetadata (TestingConfigurationMetadata.Browser, browser);
+
+      var isWebTest = browser == "NoBrowser" ? "false" : "true";
+      item.SetMetadata (TestingConfigurationMetadata.IsWebTest, isWebTest);
+
+      var databaseSystem = configurationItems[1];
+      item.SetMetadata (TestingConfigurationMetadata.DatabaseSystem, databaseSystem);
+
+      var isDatabaseTest = databaseSystem == "NoDb" ? "false" : "true";
+      item.SetMetadata (TestingConfigurationMetadata.IsDatabaseTest, isDatabaseTest);
+
+      var platform = configurationItems[2];
+      item.SetMetadata (TestingConfigurationMetadata.Platform, platform);
+
+      var use32Bit = platform.Contains ("86") ? "true" : "false";
```

Comment:
I kept thinking about arm86 architectures but you're completely right, it doesn't matter to us at all.

Reply:
Nice thinking, who knows one day could be possible. But not necessary here, dont give me nightmares.

On a different Page: We shoudl scrap this completely. Use32Bit is set in TestingConfigurations, and we should probably use that value. Are we passing AllPlatforms and stuff in here?

Reply:
I was under the impression that our new TestingConfigurations will be all that will be passed down to the MSBuild components and that we would need to refactor the existing MSBuild code to accept our new Items anyway, since we also have to refactor the ConfigurationID to be passed this way instead of being defined as a property somewhere in the MSBuild code.

Reply:
Mhmm, that makes sense, i just wasnt aware of that change. But that would mean that we dont define a list of allowed inputs anymore at all. Not sure if thats what we would need. Alternative would be to pass all the AllPlatforms/AllBrowsers etc. to the BuildTools Task to keep the information central in the BuildScript

Reply:
No, I thought the information if a build is run in 32 bit or not would not be derived from `AllPlatforms` but from the TestingConfiguration itself. And the supported platforms, browsers etc. **are** in fact being passed to the C# filtering task at the moment, which makes sense imo. Here's what the usage looks like with the new naming:

```
    <FilterTestingConfigurations
        Input="@(UnfilteredTestingConfigurations)"
        SupportedPlatforms="@(SupportedPlatforms)"
        SupportedDatabaseSystems="@(SupportedDatabaseSystems)"
        SupportedBrowsers="@(SupportedBrowsers)">
      <Output TaskParameter="Output" ItemName="TestingConfigurations"/>
    </FilterTestingConfigurations>
```

Reply:
I think we have a different understanding of SupportedPlatforms. I am talking about the AllPlatforms, which defines all possible Inputs (all Platforms, all Browsers, All databases), defined in the TestingConfiguration File (I see the possible confusion there 😬 ) As far as i understand it, SupportedPlatforms is the List of Filters for the TestOutputFiles ( aka only x64, or x64 and x86 etc.) Please correct me if i am wrong somewhere. But its probably best to discuss the Location of this information with MK